File: app/src/main/java/com/studying/countries/network/model/Currency.java
package com.studying.countries.network.model;
import android.os.Parcel;
import android.os.Parcelable;

import java.util.HashMap;
import java.util.Map;

public class Currency implements Parcelable {
    public String code;
    public String name;

    protected Currency(Parcel in) {
        code = in.readString();
        name = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(code);
        dest.writeString(name);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Currency> CREATOR = new Creator<Currency>() {
        @Override
        public Currency createFromParcel(Parcel in) {
            return new Currency(in);
        }

        @Override
        public Currency[] newArray(int size) {
            return new Currency[size];
        }
    };
}

File: app/src/main/java/com/studying/countries/network/model/Currency.kt
package com.studying.countries.network.model

import android.os.Parcel
import android.os.Parcelable

class Currency protected constructor(`in`: Parcel) : Parcelable {
    var code: String?
    var name: String?
    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(code)
        dest.writeString(name)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<Currency?> = object : Parcelable.Creator<Currency?> {
            override fun createFromParcel(`in`: Parcel): Currency? {
                return Currency(`in`)
            }

            override fun newArray(size: Int): Array<Currency?> {
                return arrayOfNulls(size)
            }
        }
    }

    init {
        code = `in`.readString()
        name = `in`.readString()
    }
}